docs/datasets.md

```diff
@@ -0,0 +1,80 @@
+# Custom Datasets in TorchTitan
+
+TorchTitan is designed to work seamlessly with most HuggingFace datasets. While we provide the C4 dataset for numerics and convergence testing, you can easily add support for your own datasets. Here's how to do it using Wikipedia as an example.
+
+## Quick Start
+
+1. Install TorchTitan from source:
+```bash
+pip install -e .
+```
```

Comment:
nit: I don't think one needs to `pip install` to use torchtitan :)

Reply:
huh interesting this wasn't obvious to me